nanoFramework.Hardware.Esp32.Rmt/ReceiverChannel.cs

```diff
@@ -13,79 +13,79 @@ namespace nanoFramework.Hardware.Esp32.Rmt
+        /// and <see cref="RmtCommand"/>s are copied into the ring buffer and availble to your code.
         /// </summary>
-        public TimeSpan ReceiveTimeout
+        public ushort IdleThreshold
         {
-            get => _receiveTimeout;
-            set => _receiveTimeout = value;
+            get => _receiverChannelSettings.IdleThreshold;
+            set
+            {
+                NativeRxSetIdleThresold(value);
+                _receiverChannelSettings.IdleThreshold = value;
+            }
         }
+
         /// <summary>
-        /// Public constructor to create receiver channel object.
+        /// Gets or sets a value indicating if the the filter is enabled. 
+        /// If enabled, the receiver will ignore pulses with widths less than specified in <see cref="FilterThreshold"/>.
         /// </summary>
-        /// <param name="gpio">The GPIO pin number that we want to use for receiving</param>
-        /// <param name="rmtBufferSize">The maximum number of RMT commands to be reserved in receiver input buffer. 
-        /// Default 100.</param>
-        public ReceiverChannel(int gpio, int rmtBufferSize = 100)
+        public bool EnableFilter
         {
-            // Default to 1MHZ tick ( 1us )
-            _clockDivider = 80;
-
-            // Initialise channel with GPIO and ring buffer size
-            _channel = NativeRxInit(gpio, rmtBufferSize);
+            get => _receiverChannelSettings.EnableFilter;
+            set
+            {
+                NativeRxEnableFilter(value, _receiverChannelSettings.FilterThreshold);
+                _receiverChannelSettings.EnableFilter = value;
+            }
         }
 
         /// <summary>
-        /// Start receiving data on channel.
+        /// Gets or sets the threshold for pulse widths to ignore when <see cref="EnableFilter"/> is set to <see langword="true"/>.
         /// </summary>
-        /// <param name="clearBuffer">Clears buffer before starting.</param>
-        public void Start(bool clearBuffer)
+        public byte FilterThreshold
         {
-            NativeRxStart(clearBuffer);
+            get => _receiverChannelSettings.FilterThreshold;
+            set
+            {
+                NativeRxEnableFilter(_receiverChannelSettings.EnableFilter, value);
+                _receiverChannelSettings.FilterThreshold = value;
+            }
         }
 
         /// <summary>
-        /// Stop receiving data on channel.
+        /// Initializes a new instance of the <see cref="ReceiverChannel"/> class.
         /// </summary>
-        public void Stop()
+        /// <param name="settings">The channel settings to use.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="settings"/> cannot be null.</exception>
+        public ReceiverChannel(ReceiverChannelSettings settings) : base(settings)
         {
-            NativeRxStop();
+            _receiverChannelSettings = settings ?? throw new ArgumentNullException(nameof(settings));
```

Comment:
In order to save flash and RAM we're not adding the name of the parameter being null. 
In this case it's obvious and already documented in the comments.
```suggestion
            _receiverChannelSettings = settings ?? throw new ArgumentNullException();
```